### PR TITLE
feat: modal component for saving any asset specified

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

https://github.com/DARPA-ASKEM/terarium/assets/28237258/1619c912-92fd-412b-869a-77983c8b62b1
 
- Created a service for saving assets `save-asset.ts` (common saving logic was consolidated: create the asset, add to current project, potentially open it)
- Created a modal component `tera-save-asset-modal.vue`  
- Right now `Model`, `Workflow` and `File` (Code) can be saved
- Removed old save modal from `tera-code.vue` as well as some unused code

My `asset` prop and the variable `newAsset` in `tera-save-asset-modal.vue`  and `save-asset.ts` accept various types. I am not too sure if this is the optimal way to go but I didn't think it was worth creating a save modal and save service for every asset type.

### CODE REVIEW

**Review by blanchco — APPROVED** [review]
looks good!

File: packages/client/hmi-client/src/page/project/components/tera-save-asset-modal.vue [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+			<template #default>
+				<form @submit.prevent>
+					<label for="new-name">What would you like to call it?</label>
+					<InputText
+						id="new-name"
+						type="text"
+						v-model="newName"
+						placeholder="Enter a unique name"
+					/>
+					<slot name="extra-input-fields" />
```
Comment (YohannParis) [inline packages/client/hmi-client/src/page/project/components/tera-save-asset-modal.vue:22]: If nobody is using it, let's remove it. We will figure about future uses later on.
Adding a `description` test area would be great here, and making it not mandatory.

File: packages/client/hmi-client/src/services/save-asset.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		logger.info(`Failed to save ${assetType}.`);
+		return;
+	}
+
+	const projectId = projectResource.activeProject.value?.id;
+	if (!projectId) {
+		logger.error(`Asset can't be saved since target project doesn't exist.`);
+		return;
+	}
+	await projectResource.addAsset(assetType, response.id, projectId);
```
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/services/save-asset.ts:53]: sure I'll do that, I was just seeing that pattern in some other components

File: packages/client/hmi-client/src/services/save-asset.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	if (!response) {
+		logger.error(`Failed to update ${assetType}.`);
+		return;
+	}
+
+	// TODO: Consider calling this refresh within the update functions in the services themselves
+	projectResource.refresh();
+
+	logger.info(`Updated ${response.name}.`);
```
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/services/save-asset.ts:106]: If we zero in on using this saving service wouldn't it be easier to use this as a wrapper for those update functions though? For these kinds of messages, etc

**Pull request merged**